### PR TITLE
Showing real deployment transaction hash

### PR DESCRIPTION
**Pull request by rzadp**
In React and in Wallet

### CODE REVIEW

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx [hunk 1]
```diff
@@ -44,20 +53,15 @@ export const Onboarding = (props: OnboardingProps) => {
...
     };
     modalService.showModal('topUpAccount', topUpProps);
     await waitForBalance();
-    modalService.showModal('waitingFor', {
-      relayerConfig,
-      action: 'Wallet creation',
-      transactionHash: '0xee9270ccdeb9fcb92b3ec509ba11ba2362ab32ba8f...'}
-    );
-    const wallet = await walletService.deployFutureWallet(ensName, gasParameters.gasPrice.toString(), gasParameters.gasToken);
+    showWalletCreationModal('Not yet available...');
```
Comment (Marik-D) [inline universal-login-react/src/ui/Onboarding/Onboarding.tsx]: The modal has an etherscan link which in this case will incorrectly navigate the user to "https://ethrscan.com/tx/Not yet available...".
Instead we should make a transaction hash optional in modal props, and make the modal use the text "Not yet available..." if `txHash === undefined`

File: universal-login-sdk/lib/core/services/WalletService.ts [hunk 2]
```diff
@@ -62,7 +62,7 @@ export class WalletService {
     return futureWallet;
   }
 
-  async deployFutureWallet(ensName: string, gasPrice: string, gasToken: string) {
+  async deployFutureWallet(ensName: string, gasPrice: string, gasToken: string, onTransactionHash?: Function) {
```
Comment (Marik-D) [inline universal-login-sdk/lib/core/services/WalletService.ts]: `onTransactionHash?: (txHash: string) => void`

File: universal-login-wallet/src/ui/react/CreateAccount/CreateAccount.tsx [hunk 3]
```diff
@@ -27,8 +28,8 @@ export const CreateAccount = () => {
     };
     modalService.showModal('topUpAccount', topUpProps);
     await waitForBalance();
-    modalService.showModal('waitingForDeploy');
-    await walletService.deployFutureWallet(name, gasParameters.gasPrice.toString(), ETHER_NATIVE_TOKEN.address);
+    showWaitingModal('Not yet available...');
```
Comment (Marik-D) [inline universal-login-wallet/src/ui/react/CreateAccount/CreateAccount.tsx]: Same as the comment above

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx (hunk 1 above)
Comment (rzadp) [reply universal-login-react/src/ui/Onboarding/Onboarding.tsx]: Good point

File: universal-login-sdk/lib/core/services/WalletService.ts (hunk 2 above)
Comment (cehali) [reply universal-login-sdk/lib/core/services/WalletService.ts]: +1

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx (hunk 1 above)
Comment (cehali) [reply universal-login-react/src/ui/Onboarding/Onboarding.tsx]: 👍

Comment (rzadp) [reply universal-login-react/src/ui/Onboarding/Onboarding.tsx]: Done now, looks like it was already implemented:
https://github.com/UniversalLogin/UniversalLoginSDK/blob/2adbe90b0664b484aecd79c309d6c1d6bd9fa971/universal-login-react/src/ui/commons/ExplorerLink.tsx#L11-L13

File: universal-login-sdk/lib/core/services/WalletService.ts (hunk 2 above)
Comment (rzadp) [reply universal-login-sdk/lib/core/services/WalletService.ts]: Done

File: universal-login-wallet/src/ui/react/CreateAccount/CreateAccount.tsx (hunk 3 above)
Comment (rzadp) [reply universal-login-wallet/src/ui/react/CreateAccount/CreateAccount.tsx]: Done

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx [hunk 4]
```diff
@@ -36,6 +36,15 @@ export const Onboarding = (props: OnboardingProps) => {
     props.onConnect && props.onConnect();
   };
 
+  const showWalletCreationModal = (transactionHash?: string) => {
+    const relayerConfig = props.sdk.getRelayerConfig();
+    modalService.showModal('waitingFor', {
+      relayerConfig,
+      action: 'Wallet creation',
+      transactionHash}
```
Comment (Marik-D) [inline universal-login-react/src/ui/Onboarding/Onboarding.tsx]:
```suggestion
      transactionHash
 }
```

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx [hunk 5]
```diff
@@ -36,6 +36,15 @@ export const Onboarding = (props: OnboardingProps) => {
...
+    );
```
Comment (Marik-D) [inline universal-login-react/src/ui/Onboarding/Onboarding.tsx]:
```suggestion
    });
```

Comment (Marik-D) [inline universal-login-react/src/ui/Onboarding/Onboarding.tsx]:
```suggestion
    });
```

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx (hunk 4 above)
Comment (Marik-D) [inline universal-login-react/src/ui/Onboarding/Onboarding.tsx]:
```suggestion
      transactionHash
```

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx [hunk 6]
```diff
@@ -36,6 +36,15 @@ export const Onboarding = (props: OnboardingProps) => {
     props.onConnect && props.onConnect();
   };
 
+  const showWalletCreationModal = (transactionHash?: string) => {
+    const relayerConfig = props.sdk.getRelayerConfig();
+    modalService.showModal('waitingFor', {
+      relayerConfig,
+      action: 'Wallet creation',
+      transactionHash
```
Comment (Marik-D) [inline universal-login-react/src/ui/Onboarding/Onboarding.tsx]:
```suggestion
      transactionHash,
```

File: universal-login-sdk/lib/core/services/WalletService.ts [hunk 7]
```diff
@@ -71,8 +71,10 @@ export class WalletService {
     const applicationWallet = {contractAddress, name: ensName, privateKey};
     this.stateProperty.set({kind: 'Deploying', wallet: applicationWallet});
 
-    const execution = await deploy(ensName, gasPrice, gasToken);
-    const deployedWallet = await execution.waitToBeSuccess();
+    const {waitToBeSuccess, waitForTransactionHash} = await deploy(ensName, gasPrice, gasToken);
+    const {transactionHash} = await waitForTransactionHash();
+    transactionHash !== null && onTransactionHash !== undefined && onTransactionHash(transactionHash);
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/core/services/WalletService.ts]: Unnecessary: `transactionHash !== null`

Comment (rzadp) [reply universal-login-sdk/lib/core/services/WalletService.ts]: Addressed by https://github.com/UniversalLogin/UniversalLoginSDK/pull/1330

**Pull request merged**